.buildkite/hooks/pre-command

```diff
@@ -10,6 +10,10 @@ if [[ "${BUILDKITE_LABEL:-}" == *"Pipeline upload"* || "${BUILDKITE_LABEL:-}" ==
   fi
 fi
 
+if [[ "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-packetbeat" && "$BUILDKITE_STEP_KEY" == *"system-tests"* ]]; then
+    source "$BUILDKITE_BUILD_CHECKOUT_PATH/.buildkite/scripts/gcp_auth.sh"
```

Comment:
As a general note, we are trying to not have a lot of "magic invocations" in pre-command as they reduce discoverability when trying to reproduce scenarios (e.g. see https://docs.elastic.dev/ingest-dev-docs/beats/beats-ci#reproducing-pull-request-failures). This means a little more duplication in favor of having a contained list of CI commands to execute.

If it's not a problem and you don't see the need to have many more steps in the future that duplicate this I'd recommend leaving the source command in the steps where it was. WDYT?

Reply:
Thanks for porting the script to powershell, it seems to work now ❤️